### PR TITLE
Auto-update dependencies: output info in machine-readable format

### PR DESCRIPTION
**Pull request by jodersky**
Mill has a convenient task to check for the availability of newer versions of dependencies `mill.scalalib.Dependency/updates`.

Unfortunately this task only prints the output (in human readable form), but does not persist it in a machine-readable format. This makes it very difficult to reuse the output in other tasks, essentially requiring regex parsing of strings.

The changes proposed here change the `updates` task to expose the underlying data structures representing the upgrade path. An additional task, `showUpdates` is added, implementing the original, human readable, output of updates.

### CODE REVIEW

File: scalalib/src/dependency/versions/ModuleDependenciesVersions.scala [hunk 1]
```diff
@@ -3,10 +3,10 @@ package mill.scalalib.dependency.versions
 import mill.scalalib.JavaModule
 
 private[dependency] final case class ModuleDependenciesVersions(
-    module: JavaModule,
```
Comment (lefou) [inline scalalib/src/dependency/versions/ModuleDependenciesVersions.scala]: What's the motivation for this change?

File: scalalib/src/dependency/versions/ModuleDependenciesVersions.scala [hunk 2]
```diff
@@ -3,10 +3,10 @@ package mill.scalalib.dependency.versions
...
+    modulePath: String,
     dependencies: Seq[DependencyVersions])
 
 private[dependency] final case class DependencyVersions(
-    dependency: coursier.Dependency,
```
Comment (lefou) [inline scalalib/src/dependency/versions/ModuleDependenciesVersions.scala]: Wouldn't be `coursier.Dependency` the better type for later inspection of possible updates?

File: scalalib/src/dependency/versions/ModuleDependenciesVersions.scala (hunk 1 above)
Comment (jodersky) [reply scalalib/src/dependency/versions/ModuleDependenciesVersions.scala]: The issue is that `JavaModule` is not picklable, hence it can't be used in the updates task's result

File: scalalib/src/dependency/versions/ModuleDependenciesVersions.scala (hunk 2 above)
Comment (jodersky) [reply scalalib/src/dependency/versions/ModuleDependenciesVersions.scala]: yes indeed. I wasn't sure if coursier should be exposed by mill, so I used a string instead. If it's ok to leak coursier, then I'd be happy to use that instead

Comment (lefou) [reply scalalib/src/dependency/versions/ModuleDependenciesVersions.scala]: I guess, mill already leaks `coursier.Dependency` as it's an public property of `mill.scalalib.Dep`. So I think it would be ok expose `coursier.Dependency` here. We currently have no other abstraction / API for managed dependencies.

File: scalalib/src/dependency/versions/ModuleDependenciesVersions.scala (hunk 1 above)
Comment (lefou) [reply scalalib/src/dependency/versions/ModuleDependenciesVersions.scala]: Yeah, that makes sense.

**Comment by jodersky** [comment]
@lefou, I addressed your comments

File: scalalib/src/Dependency.scala [hunk 3]
```diff
@@ -18,6 +18,10 @@ object Dependency extends ExternalModule {
         allowPreRelease)
     }
 
+  def showUpdates(ev: Evaluator, allowPreRelease: Boolean = false) = T.command{
```
Comment (lefou) [inline scalalib/src/Dependency.scala]: Can you please add some ScalaDoc to the new command. It would be great, if you could also update/add the comment of the changed `updates` command.

**Comment by lefou** [comment]
Thank you!
